src/main/webapp/javascript/user-auth.js

```diff
@@ -0,0 +1,96 @@
+var auth2;
+
+function getId() {
+  return auth2.currentUser.get().getId();
+}
+
+/**
+ * loads the auth2 library for the g-signin API, and then initializes
+ * the googleAuth object.
+ * passed as jquery parameter, executed by API script.
+ */
+function onStart() {
+  console.log("onstart");
+  gapi.load('auth2', initSigninV2);
+}
+
+/**
+ * initializes google authentication API and sets event listeners.
+ */
+function initSigninV2() {
+  auth2 = gapi.auth2.init({
+    client_id : '1034390229233-u07o0iaas2oql8l4jhe7fevpfsbrtv7n.apps.googleusercontent.com'
+  });
+
+  gapi.signin2.render('g-signin-container');
+
+  auth2.currentUser.listen(function(newGoogleUser) {
+    insertUserInfo(newGoogleUser);
+  });
+
+  auth2.isSignedIn.listen(function(signedIn) {
+    if (signedIn) {
+      console.log('enable dropdown');
+      enableUserDropdown();
+    } else {
+      console.log('disable dropdown');
+      disableUserDropdown();
+    }
+  });
+}
+
+
+
+/**
+ * signs our current user from g-signin API.
+ */
+function signOut() {
+  auth2.signOut().then(function () {
+    console.log('User signed out.');
+  });
+}
+
+/**
+ * To be run in currentUser listener, activates whenever current user changes.
+ *
+ * @param {GoogleUser} current user to provide profile info.
+ */
+function insertUserInfo(googleUser) {
+  // insert user name into user-page anchor
+  let text = googleUser.getBasicProfile().getGivenName() + '\'s Page';
```

Comment:
Does this still create an error if the user isn't initially signed in like we saw yesterday? If so, you may want to do a check on `googleUser` and `googleUser.getBasicProfile()`.